contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+        _withdraw(_recipient, _amount);
+    }
+
+    /**
+     * @notice Withdraw all supported assets from platform and
+     * sends to the OToken's Vault.
+     */
+    function withdrawAll()
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _removeAllLiquidity();
+
+        uint256 balance = IERC20(WETH).balanceOf(address(this));
+        if (balance > 0) {
+            _withdraw(vaultAddress, balance);
+        }
+        // burn remaining OETHb
+        _burnOETHbOnTheContract();
+    }
+
+    function _withdraw(
+        address _recipient,
+        uint256 _amount
+    ) internal {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+
+        IERC20(WETH).safeTransfer(_recipient, _amount);
+        emit Withdrawal(WETH, address(0), _amount);
+    }
+
+
+    /**
+     * @dev Collect the AERO token from the gauge 
+     */
+    function _collectRewardTokens() internal override {
+        // TODO do other stuff
+        super._collectRewardTokens();
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset) public view override returns (bool) {
+        return _asset == WETH;
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / pTokens
+            We need to give the Aerodrome pool approval to transfer the
+            asset.
+     * @param _asset Address of the asset to approve
+     * @param _pToken Address of the pToken
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(_asset).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /**
+     * @dev Approve the spending of all assets 
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        IERC20(WETH).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(positionManager), type(uint256).max);
+        IERC20(WETH).safeApprove(address(swapRouter), type(uint256).max);
+        IERC20(OETHb).safeApprove(address(swapRouter), type(uint256).max);
+    }
+
+    /***************************************
+            Balances and Fees
+    ****************************************/
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return underlyingAssets;
```

Comment:
Add a `asset == WETH` check here. Otherwise, we would have this contract reporting incorrect values when someone (or analytics tools) query it with some other address

Reply:
good point, fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/a240c378d9d59ac47edc0e003feadf909352236c